Fix issue where genre wasn't showing and year is empty

fernandog opened the pull request:
Its not in my repo so other can help.
Seems the empty year is was causing the tag not showing

 I also can't understand why there's another minor empty tag

In this first two images I change html to show genres:

Working show:
![1](https://cloud.githubusercontent.com/assets/2620870/11717754/9c187e4e-9f39-11e5-9514-5afd43cd82b5.png)

Before:
![2](https://cloud.githubusercontent.com/assets/2620870/11717755/9c746fc4-9f39-11e5-89fd-de05679c7cda.png)

After:
![image](https://cloud.githubusercontent.com/assets/2620870/11718585/d0f28764-9f3d-11e5-81a1-855ae0b3a655.png)

fernandog commented:
@SickRage/collaborators

OmgImAlexis commented on gui/slick/views/displayShow.mako:
```diff
@@ -126,8 +126,10 @@
+    % if 'year' in show.imdb_info and len(show.imdb_info['year']) == 4:
                 <span>(${show.imdb_info['year']}) - ${show.imdb_info['runtimes']} minutes - </span>
+    % else:
+                <span>${show.imdb_info['runtimes']} minutes</span>
```
Shouldn't this

````
% if 'year' in show.imdb_info and len(show.imdb_info['year']) == 4:
    <span>(${show.imdb_info['year']}) - ${show.imdb_info['runtimes']} minutes - </span>
% else:
    <span>${show.imdb_info['runtimes']} minutes</span>
% endif
````

be this?

````
<span>
% if 'year' in show.imdb_info and len(show.imdb_info['year']) == 4:
    (${show.imdb_info['year']}) - 
% endif
${show.imdb_info['runtimes']} minutes</span>
````

fernandog replied:
@OmgImAlexis same result as image "After"

miigotu replied:
`if 'year' in show.indb_info and show.imdb_info['year']:`

miigotu commented on gui/slick/views/displayShow.mako:
```diff
@@ -139,7 +141,7 @@
 
             <div id="tags">
                 <ul class="tags">
-                    % if not show.imdbid and show.genre:
```
`% if show.imdbid and show.genre:`

fernandog replied:
I added that. but genre comes from trakt or imdb?

fernandog commented:
I added @miigotu and @OmgImAlexis changes but still:

${show.genre[1:-1].split('|')} = [u'Action', u'Drama']

any clue wtf is going on?

![image](https://cloud.githubusercontent.com/assets/2620870/11722376/7935a0a8-9f4f-11e5-8ad2-f14dac638a77.png)

fernandog commented:
Seems IMDB genre that is empty. not Trakt.....

Thraxis commented on gui/slick/views/displayShow.mako:
```diff
@@ -139,7 +142,7 @@
-                    % if not show.imdbid and show.genre:
+                    % if show.imdbid and show.genre:
                         % for genre in show.genre[1:-1].split('|'):
                             <a href="${anon_url('http://trakt.tv/shows/popular/?genres=', genre.lower())}" target="_blank" title="View other popular ${genre} shows on trakt.tv."><li>${genre}</li></a>
```
Can add a check here:

```python
                         % for genre in show.genre[1:-1].split('|'):
                               % if genre.strip():
                             <a href="${anon_url('http://trakt.tv/shows/popular/?genres=', genre.lower())}" target="_blank" title="View other popular ${genre} shows on trakt.tv."><li>${genre}</li></a>                
```

duramato commented:
And wab the issue @OmgImAlexis was talking of the imdb genre differing from trakt ?

fernandog commented:
we were checking YEAR to display GENRE. makes no sense.

we can display genres from indexer or imdb.
imdb genre has preference.

fernandog commented:
@OmgImAlexis @duramato now imdb genres are used to create imdb genre link. no issues anymore

miigotu commented on gui/slick/views/displayShow.mako:
```diff
@@ -139,14 +142,14 @@
                         % endfor
                     % endif
-                    % if 'year' in show.imdb_info:
+                    % if show.imdb_info['genres']:
```
`if 'genres' in show.imdb_info and show.imdb_info['genres']:`

fernandog commented:
@miigotu done
